### ISSUE TITLE
better toml handling

### ISSUE THREAD

**Issue by fdncred** [nushell/engine-q#450]
This is taken from [nushell's repo](https://github.com/nushell/nushell/issues/2132) but instead of being for better toml handling of the configuration system, it's for `from toml` and `to toml`, since we're using nu script for the config system. The real key to "better toml handing" is really just using the [toml_edit](https://crates.io/crates/toml_edit) crate.

**Comment by kaleidawave** [comment]
Awesome, thanks for opening this issue. I don't use nushell a lot but fixing this use case would make me use it more. Will rewrite what I wrote in old issue:

I want to modify some toml files (change the values under keys). These toml are configurations for another ecosystem (this is not to do with the `config` command).

Currently with `open x.toml | modify_thing | save x.toml` the saved toml file has comments stripped and its keys and sections(/tables) sorted to be in alphabetic order. These are developer facing files, so reordering entries and removing comments is a negative.

Also does reordering happen for JSON modification...? Being able to modify these files directly from the term is super helpful but retaining entry order, non-data and their formatting (indent width,) is necessary

**Comment by fdncred** [comment]
If our json tools were wrote with our nu-json crate then I believe ordering should be maintained as well as handling json comments.

**Comment by GabrielBG0** [comment]
well, as no one claimed this I can get it

**Comment by fdncred** [comment]
you got it. have fun!

**Comment by onthebridgetonowhere** [comment]
Bumping this up, seems a good issue to fix. 

@GabrielBG0 - you managed to fix this problem?

**Comment by GabrielBG0** [comment]
No, I started to some changes but I think I bit more than I could chew here.
I'm afraid I lack both Rust and project knowledge to be able to fix it.
But if someone wants to check out what I did until now and give it a try u can finde the code [here](https://github.com/GabrielBG0/engine-q/commit/e7b45fdeb775c2b2d543a77bb65189f2ba61e934)
